assets/src/edit-story/components/library/panes/text/textSets/textSet.js

```diff
@@ -44,42 +52,123 @@ const TextSetItem = styled.button`
   cursor: pointer;
 `;
 
+const DragContainer = styled.div`
+  position: absolute;
+  top: 0;
+  left: 0;
+  z-index: -1;
+  width: ${({ width }) => width}px;
+  height: ${({ height }) => height}px;
+  background-color: ${({ theme }) => rgba(theme.colors.bg.white, 0.2)};
+`;
+
 function TextSet({ elements }) {
-  const insertTextSet = useInsertTextSet();
+  const { insertTextSet } = useLibrary((state) => ({
+    insertTextSet: state.actions.insertTextSet,
+  }));
+
+  const elementRef = useRef();
+
+  const { canvasPageSize } = useLayout(({ state }) => ({
+    canvasPageSize: state.canvasPageSize,
+  }));
+
+  const handleDragStart = useCallback(
+    (e) => {
+      const { x, y } = e.target.getBoundingClientRect();
+      const offsetX = e.clientX - x;
+      const offsetY = e.clientY - y;
+
+      e.dataTransfer.setDragImage(elementRef.current, offsetX, offsetY);
+      e.dataTransfer.setData(
+        'textset',
+        JSON.stringify({
+          grabOffsetX: -offsetX,
+          grabOffsetY: -offsetY,
+          elements,
+        })
+      );
+    },
+    [elements]
+  );
+
+  const { textSetHeight, textSetWidth } = elements[0];
+  const { width: pageWidth, height: pageHeight } = canvasPageSize;
+  const dragWidth = dataToEditorX(textSetWidth, pageWidth);
+  const dragHeight = dataToEditorY(textSetHeight, pageHeight);
 
   return (
-    <TextSetItem
-      role="listitem"
-      aria-label={__('Insert Text Set', 'web-stories')}
-      onClick={() => insertTextSet(elements)}
-    >
-      {elements.map(
-        ({
-          id,
-          content,
-          previewOffsetX,
-          previewOffsetY,
-          textSetWidth,
-          textSetHeight,
-          ...rest
-        }) => (
-          <DisplayElement
-            previewMode
-            key={id}
-            element={{
+    <>
```

Comment:
I think we should add an actual DOM wrapper node here and make that `role="listitem"` rather than the current result with the list being a mix of invisible drag containers and actual list item buttons.

Just something like `<div role="listitem">`, but probably by going through `styled.div`.